Source/SwiftLintFramework/Extensions/Configuration+Merging.swift

```diff
@@ -43,10 +43,69 @@ extension Configuration {
         return self
     }
 
-    // Currently merge simply overrides the current configuration with the new configuration.
-    // This requires that all configuration files be fully specified. In the future this should be
-    // improved to do a more intelligent merge allowing for partial nested configurations.
+    private struct HashableRule: Hashable {
+        let rule: Rule
+
+        static func == (lhs: HashableRule, rhs: HashableRule) -> Bool {
+            // Don't use `isEqualTo` in case its internal implementation changes from
+            // using the identifier to something else, which could mess up with the `Set`
+            return type(of: lhs.rule).description.identifier == type(of: rhs.rule).description.identifier
+        }
+
+        var hashValue: Int {
+            return type(of: rule).description.identifier.hashValue
+        }
+    }
+
+    private func mergingRules(with configuration: Configuration) -> [Rule] {
+        switch configuration.rulesMode {
+        case .allEnabled:
+            // Technically not possible yet as it's not configurable in a .swiftlint.yml file,
+            // but implemented for completeness
+            return configuration.rules
```

Comment:
I'd agree, but the fact that we're not merging the individual rule's configurations yet leads to this code for now.
I can update the comment that the implementation needs to change in order to properly support that feature, if it ever becomes supported